experimental/golden_response_checker.go

```diff
@@ -0,0 +1,122 @@
+package experimental
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+// CheckGoldenDataResponse will verify that the stored file matches the given data.DataResponse
+// when the updateFile flag is set, this will both add errors to the response and update the saved file
+func CheckGoldenDataResponse(path string, dr *backend.DataResponse, updateFile bool) error {
+	saved, err := readGoldenFile(path)
+	if err != nil {
+		err = fmt.Errorf("error reading golden file:  %s\n%s", path, err.Error())
+	} else {
+		if diff := cmp.Diff(saved.Error, dr.Error); diff != "" {
+			err = fmt.Errorf("errors mismatch %s (-want +got):\n%s", path, diff)
+		}
+		if diff := cmp.Diff(len(saved.Frames), len(dr.Frames)); diff != "" {
+			err = fmt.Errorf("Frame count mismatch (-want +got):\n%s", diff)
+		} else {
+			// Check each frame
+			for idx, frame := range dr.Frames {
+				expectedFrame := saved.Frames[idx]
+				if diff := cmp.Diff(expectedFrame, frame, data.FrameTestCompareOptions()...); diff != "" {
+					err = fmt.Errorf("Frame[%d] mismatch (-want +got):\n%s", idx, diff)
+				}
+			}
+		}
+	}
+
+	if err != nil && updateFile {
+		_ = writeGoldenFile(path, dr)
+		log.Printf("golden file updated: %s\n", path)
+	}
+	return err
+}
+
+const binaryDataSection = "\n====== TEST DATA RESPONSE (arrow base64) ======\n"
+
+func readGoldenFile(path string) (*backend.DataResponse, error) {
```

Comment:
I think bufio.Scanner will make this cleaner https://golang.org/pkg/bufio/#example_Scanner_lines

Reply:
:+1: yes, much better!